File: frontend/static/script/models/locations/Outpost.js
"use strict";
import { Settlement } from "../blueprints/Settlement.js";



export class Outpost extends Settlement{
    constructor(settlement_id, settlement_type, name, resources, buildings, population, free_workers, free_builders) {
        super(settlement_id, settlement_type, name, resources, buildings, population, free_workers, free_builders);

    }
}

File: backend_refactoring/CSharpApi/Models/BluePrints/Locations/Outpost.cs

using CSharpApi.Models.BluePrints.Beings;
using CSharpApi.Models.BluePrints.BuildingTypes;

namespace CSharpApi.Models.BluePrints.Locations
{
    public class Outpost : Settlement
    {
        public Outpost(string settlementName, float gold, float food, float wood, float iron, string settlementType = "city", List<BuildingState> buildings = null, List<Population> population = null) : base(settlementName, settlementType, gold, food, wood, iron, buildings ?? [], population ?? [])
        {
        }
    }
}
